pkg/roachprod/prometheus/prometheus.go

```diff
@@ -0,0 +1,488 @@
+			curl -fsSL https://storage.googleapis.com/cockroach-fixtures/prometheus/prometheus-2.27.1.linux-amd64.tar.gz | tar zxv --strip-components=1`,
+	); err != nil {
+		return nil, err
+	}
+	// create and upload prom config
+	yamlCfg, err := makeYAMLConfig(cfg.ScrapeConfigs)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := cfg.PromNodeHandle.PutString(
+		ctx,
+		l,
+		yamlCfg,
+		"/tmp/prometheus/prometheus.yml",
+		0644,
+	); err != nil {
+		return nil, err
+	}
+
+	// Start prometheus as systemd.
+	if err := c.Run(
+		ctx,
+		l,
+		os.Stdout,
+		os.Stderr,
+		cfg.PrometheusNode,
+		"start-promethius",
+		`cd /tmp/prometheus &&
+sudo systemd-run --unit prometheus --same-dir \
+	./prometheus --config.file=prometheus.yml --storage.tsdb.path=data/ --web.enable-admin-api`,
+	); err != nil {
+		return nil, err
+	}
+
+	if cfg.Grafana.Enabled {
+		// Install Grafana.
+		if err := c.RepeatRun(ctx, l,
+			os.Stdout,
+			os.Stderr, cfg.PrometheusNode, "install grafana",
+			`sudo apt-get install -qqy apt-transport-https &&
+sudo apt-get install -qqy software-properties-common wget &&
+wget -q -O - https://packages.grafana.com/gpg.key | sudo apt-key add - &&
+echo "deb https://packages.grafana.com/enterprise/deb stable main" | sudo tee -a /etc/apt/sources.list.d/grafana.list &&
+sudo apt-get update -qqy && sudo apt-get install -qqy grafana-enterprise && sudo mkdir -p /var/lib/grafana/dashboards`,
+		); err != nil {
+			return nil, err
+		}
+
+		// Provision local prometheus instance as data source.
+		if err := c.RepeatRun(ctx, l,
+			os.Stdout,
+			os.Stderr, cfg.PrometheusNode, "permissions",
+			`sudo chmod 777 /etc/grafana/provisioning/datasources /etc/grafana/provisioning/dashboards /var/lib/grafana/dashboards`,
+		); err != nil {
+			return nil, err
+		}
+		if err := cfg.PromNodeHandle.PutString(ctx, l, `apiVersion: 1
+
+datasources:
+  - name: prometheusdata
+    type: prometheus
+    uid: localprom
+    url: http://localhost:9090
+`, "/etc/grafana/provisioning/datasources/prometheus.yaml", 0644); err != nil {
+			return nil, err
+		}
+
+		if err := cfg.PromNodeHandle.PutString(ctx, l, `apiVersion: 1
+
+providers:
+ - name: 'default'
+   orgId: 1
+   folder: ''
+   folderUid: ''
+   type: file
+   options:
+     path: /var/lib/grafana/dashboards
+`, "/etc/grafana/provisioning/dashboards/cockroach.yaml", 0644); err != nil {
+			return nil, err
+		}
+
+		for idx, u := range cfg.Grafana.DashboardURLs {
+			cmd := fmt.Sprintf("curl -fsSL %s -o /var/lib/grafana/dashboards/%d.json", u, idx)
+			if err := c.Run(ctx, l, os.Stdout, os.Stderr, cfg.PrometheusNode, "download dashboard",
+				cmd); err != nil {
+				l.PrintfCtx(ctx, "failed to download dashboard from %s: %s", u, err)
+			}
+		}
+
+		// Start Grafana. Default port is 3000.
+		if err := c.Run(ctx, l, os.Stdout, os.Stderr, cfg.PrometheusNode, "start grafana",
+			`sudo systemctl restart grafana-server`); err != nil {
+			return nil, err
+		}
+	}
+
+	p := &Prometheus{Config: cfg}
```

Comment:
any tips for spitting out some nice urls to take the user to some dashboards? Currently it spits out the text below. I tried executing these on my mac, but the directions are for linux, I believe. 

```
Not creating home directory `/usr/share/grafana'.
### NOT starting on installation, please execute the following statements to configure grafana to start automatically using systemd
 sudo /bin/systemctl daemon-reload
 sudo /bin/systemctl enable grafana-server
### You can start grafana-server by executing
 sudo /bin/systemctl start grafana-server
```

Reply:
Just spitting out the external ip:3000 would be a good start? RIght now you need to login admin/admin and then navigate to dashboards -> browse, I believe grafana has an option to disable login (i.e. everyone is admin), we could enable that (but wouldn't worry about that just yet).